msyncd/AccountsHelper.cpp

```diff
@@ -186,7 +189,7 @@ void AccountsHelper::slotAccountEnabledChanged(const QString &serviceName, bool
             foreach(SyncProfile *profile, profiles)
             {
                 // See if the service name matches
-                if(serviceName == profile->name())
+                if(serviceName == profile->key(REMOTE_SERVICE_NAME))
                 {
```

Comment:
In case a specific service not the "global" account get enabled we need to check which profiles are related to this service and enable/disable it. The profile name is not the same as the service name. For example my profiles created by the account creation has names like "727230507.xml". The only profile that I have with the same name as the service is the "template" profile used to create new ones. As you can see in the function "createProfileForAccount" it will use the profile as template and create a new one we a new name. 
I do not see the case where the profile-name is the same as the service name.